internal/profile/install_profile.go

```diff
@@ -0,0 +1,212 @@
+	}
+	if newExists {
+		return errors.Errorf("profile %s already exists", newProfile.profileName)
+	}
+	os.Mkdir(newProfile.ProfilePath, 0755)
+
+	newProfile.updateFileBodies(fromProfile.configFiles)
+	return newProfile.writeProfileResources()
+
+}
+
+// LoadProfileFromDefaultLocation loads an existing profile from the default elastic-package config dir
+func LoadProfileFromDefaultLocation(profileName string) (*ConfigProfile, error) {
+
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding stack dir location")
+	}
+
+	return LoadProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfileFromDefaultLocation deletes a profile from the default elastic-package config dir
+func DeleteProfileFromDefaultLocation(profileName string) error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return DeleteProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfile deletes a given config profile.
+func DeleteProfile(elasticPackagePath string, profileName string) error {
+
+	if profileName == DefaultProfile {
+		return errors.New("cannot remove default profile")
+	}
+
+	pathToDelete := filepath.Join(elasticPackagePath, profileName)
+
+	return os.RemoveAll(pathToDelete)
+
+}
+
+// PrintProfilesFromDefaultLocation lists known packages in the default elastic-package install dir
+func PrintProfilesFromDefaultLocation() error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return PrintProfiles(loc.StackDir())
+}
+
+// PrintProfiles lists known profiles
+func PrintProfiles(elasticPackagePath string) error {
+
+	profiles, err := FetchAllProfiles(elasticPackagePath)
+	if err != nil {
+		return errors.Wrap(err, "error fetching profiles")
+	}
+
+	//print later, after we've run into any errors, to avoid some ugly-half printing if something fails halfway through
+	header := []string{"Name", "Date Created", "User", "elastic-package version", "Path"}
+	for _, headerVal := range header {
+		fmt.Printf("%-30s ", headerVal)
+	}
+	fmt.Printf("\n")
+	for _, iter := range profiles {
+		fmt.Printf("%-30s %-30s %-30s %-30s %-30s\n", iter.Name, iter.DateCreated.Format(time.RFC3339), iter.User, iter.Version, iter.From)
+	}
+
+	return nil
+}
+
+// FetchAllProfiles returns a list of profile values
+func FetchAllProfiles(elasticPackagePath string) ([]Metadata, error) {
+	dirList, err := ioutil.ReadDir(elasticPackagePath)
+	if err != nil {
+		return []Metadata{}, errors.Wrapf(err, "error reading from directory %s", elasticPackagePath)
+	}
+
+	var profiles []Metadata
+	// TODO: this should read a profile.json file or something like that
+	for _, item := range dirList {
+		if !item.IsDir() {
+			continue
+		}
+		profile, err := LoadProfile(elasticPackagePath, item.Name())
+		if err == ErrNotAProfile {
```

Comment:
Consider using [`errors.Is`](https://blog.golang.org/go1.13-errors#TOC_3.2.) here. It'll be more future-proof, in case `err` here ever decides to wrap `ErrNotAProfile`.

Reply:
Ah, didn't even know that existed!